flocker/node/_change.py

```diff
@@ -117,14 +119,19 @@ def in_parallel(changes):
     The order in which execution of the changes is started is unspecified.
     Comparison of the resulting object disregards the ordering of the changes.
 
-    @param changes: A sequence of ``IStateChange`` providers.
+    :param changes: A sequence of ``IStateChange`` providers.
+    :param timedelta sleep_when_empty: Sleep value for returned ``NoOp``
+        if no changes are given.
 
-    @return: ``IStateChange`` provider that will run given changes in
+    :return: ``IStateChange`` provider that will run given changes in
         parallel, or ``NoOp`` instance if changes are empty or all
-        ``NoOp``.
+        ``NoOp``. In former case sleep will be ``sleep_when_empty``, in
+        latter the minimum sleep of the ``NoOp`` instances.
     """
-    if all(c == NoOp() for c in changes):
-        return NoOp()
+    if all(isinstance(c, NoOp) for c in changes):
+        sleep = (min(c.sleep for c in changes) if changes
+                 else sleep_when_empty)
+        return NoOp(sleep=sleep)
```

Comment:
:question: alternatively, could add NoOp(sleep=sleep_when_empty) to the list of changes and spare the `if`.